tfx/examples/bert/utils/bert_tokenizer_utils.py

```diff
@@ -0,0 +1,187 @@
+    self._model_link = model_link
+    self._find_special_tokens()
+
+  def _find_special_tokens(self):
+    """Find the special token ID's for [CLS] [PAD] [SEP]
+
+    Since each Bert model is trained on different vocaburary, it's important
+    to find the special token index pertaining to that model.
+    Since in Transform, tensorflow_hub.KerasLayer loads a symbolic tensor, turn
+    on eager mode to get the actual vocab_file location.
+    """
+
+    with eager_mode():
+      model = hub.KerasLayer(self._model_link)
+      vocab = model.resolved_object.vocab_file.asset_path.numpy()
+      f = open(vocab, 'r')
+      self._sep_id = None
+      self._cls_id = None
+      self._pad_id = None
+      lines = f.read().split('\n')
+      for i, line in enumerate(lines):
+        if line == _PAD:
+          self._pad_id = i
+        elif line == _CLS:
+          self._cls_id = i
+        elif line == _SEP:
+          self._sep_id = i
+        if (self._pad_id is not None and
+            self._cls_id is not None and
+            self._sep_id is not None):
+          break
+
+  def tokenize_single_sentence(
+      self,
+      sequence,
+      max_len=128,
+      add_cls=True,
+      add_sep=True,
+      add_pad=True):
+    """Tokenize a single sentence according to the vocab used by the Bert model.
+
+    Add special tokens according to config.
+
+    Args:
+      sequence: Tensor of shape [batch_size, 1].
+      max_len: The number of tokens after padding and truncating.
+      add_cls: Whether to add CLS token at the front of each sequence.
+      add_sep: Whether to add SEP token at the end of each sequence.
+      add_pad: If not add_pad, would return ragged tensor instead of tensors
+        input_mask and segment_id would be none.
+
+    Returns:
+      word_id: Tokenized sequences [batch_size, max_len].
+      input_mask: Mask padded tokens [batch_size, max_len].
+      segment_id: Distinguish multiple sequences [batch_size, max_len].
+    """
+    model = hub.KerasLayer(self._model_link)
+    vocab_file_path = model.resolved_object.vocab_file.asset_path
+    tokenizer = text.BertTokenizer(vocab_file_path, token_out_type=tf.int64)
+    word_id = tokenizer.tokenize(sequence)
+    # Tokenizer default puts tokens into array of size 1. merge_dims flattens it
+    word_id = word_id.merge_dims(1, 2)
+    if add_cls:
+      cls_token = tf.fill(
+          [tf.shape(sequence)[0], 1],
+          tf.constant(self._cls_id, dtype=tf.int64))
+
+      word_id = tf.concat([cls_token, word_id], 1)
+
+    if add_sep:
+      sep_token = tf.fill(
+          [tf.shape(sequence)[0], 1],
+          tf.constant(self._sep_id, dtype=tf.int64))
+
+      word_id = word_id[:, :max_len-1]
+      word_id = tf.concat([word_id, sep_token], 1)
+
+    if not add_pad:
+      return word_id, None, None
+
+    word_id = word_id.to_tensor(
+        default_value=tf.constant(self._pad_id, dtype=tf.int64))
+
+    word_id = tf.pad(
+        word_id,
```

Comment:
you can specify a shape in to_tensor(), in this case shape=[None, max_len]

Reply:
thx! I totally missed that.